app/controllers/application_controller.rb

```diff
@@ -1,2 +1,19 @@
 class ApplicationController < ActionController::API
+  before_filter :set_headers
+
+  def index
+    render nothing: true
+  end
+
+  def set_headers
+    if request.headers["HTTP_ORIGIN"]
+      # if request.headers["HTTP_ORIGIN"] && /^https?:\/\/(.*)\.some\.site\.com$/i.match(request.headers["HTTP_ORIGIN"])
+      headers['Access-Control-Allow-Origin'] = request.headers["HTTP_ORIGIN"]
+      headers['Access-Control-Expose-Headers'] = 'ETag'
+      headers['Access-Control-Allow-Methods'] = 'GET, POST, PATCH, PUT, DELETE, OPTIONS, HEAD'
+      headers['Access-Control-Allow-Headers'] = '*,x-requested-with,Content-Type,If-Modified-Since,If-None-Match,Auth-User-Token'
+      headers['Access-Control-Max-Age'] = '86400'
+      headers['Access-Control-Allow-Credentials'] = 'true'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.